src/pages/index.js

```diff
@@ -6,121 +6,415 @@ import Section from "../components/Section.js";
+  .then((result) => {
+    userInfo = result; //store info in const
+    console.log(userInfo);
+    return api.getInitialCards();
+  })
+  .then((initialCards) => {
+    console.log(initialCards);
+    // //console.log(cardData);
+    // return userInfo._id;
+  })
+  // .then((card) => {
+  //   console.log(card);
+  //   console.log()
+  // })
+  .catch((err) => {
+    console.error(err); 
+    console.error("")
+  });
+
+  
+  api.getUserInfo()
+    .then((cardData) => {
+    editUserInfo.setUserAvatarInfo({ avatar: cardData.avatar });
+  });
+
+  
+
+api.getUserInfo()
+  .then((cardData) => {
+    editUserInfo.setUserInfo({name: cardData.name, description: cardData.about, avatar: cardData.avatar});
+  });
+
+
+
+
+  function handleEditProfileFormSubmit(cardData) {
+    editProfilePopup.submitButtonLoadingState(false); 
+  //   profileTitleInput.value = userInfo.name;   // Pre-fill with current name
+  // profileDescriptionInput.value = userInfo.about; 
+    
+    const name = profileTitleInput.value;
+    const about = profileDescriptionInput.value;
+
+    api
+    .editProfile({name, about})
+    .then((result) => {
+      console.log(result);
+      editUserInfo.setUserInfo({name: result.name, about: result.about});
+      editProfileSubmitButton.textContent = "Saving..."
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+      console.error("Changes to profile submission unsuccessful. Error."); 
+    })
+    .finally(() => {
+      editProfilePopup.submitButtonLoadingState(true); 
+      
+    });
+  }
+  
+
+  editProfileFormValidator.enableValidation();
+  
+  
+
+
+  function handleProfileUpdateAvatarFormSubmit() {
+
+    updateAvatarPopup.submitButtonLoadingState(false);
+
+    const avatar = profileAvatarUrlInput.value;
+
+    api.editProfileAvatar({avatar})
+    .then((result) => {
+      console.log(result);
+      editUserInfo.setUserAvatarInfo({avatar: result.avatar});
+      updateAvatarPopup.close();
+  console.log("Avatar has been changed to the following avatar", avatar)
+    })
+    .catch((err) => {
+      console.error(err);
+      console.error("Profile Avatar edit api unsuccessful. Error")
+    }) 
+    .finally(() => {
+      updateAvatarPopup.submitButtonLoadingState(true); 
+    });
+  }
+
+
+
+
+//CARDS
+let cardSection;
+
+api
+  .getInitialCards()
+  .then((result) => {
+    console.log(result);
+    cardSection = new Section(
+      {
+        items: result,
+        renderer: createCard,
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+    return result;
+  })
+  .catch((err) => {
+    console.error(err); 
+    console.error("Card retrieval unsuccessful. Error");
+  });
+
+
+// TODO call api.getUserInfo
+// in the .then(), set the user info
+
 // FUNCTIONS
 
+
 function handleImageClick(cardData) {
   modalImage.alt = cardData.name;
   modalImage.src = cardData.link;
   modalText.textContent = cardData.name;
   cardPreviewPopup.open(cardData);
```

Comment:
```js

  modalImage.alt = cardData.name;
  modalImage.src = cardData.link;
  modalText.textContent = cardData.name;
```

Please, delete it

You have `cardPreviewPopup.open(cardData);` for that from now on


```js
function handleImageClick(cardData) {
  cardPreviewPopup.open(cardData);
}

```